katcp/core.py

```diff
@@ -97,6 +97,24 @@ def wrapped_coro(self, *args, **kwargs):
     return wrapped_coro
 
 def log_future_exceptions(logger, f, ignore=()):
+    """Log any exceptions set to a future
+
+    Parameters
+    ----------
+    logger : logging.Logger instance
+        logger.exception(...) is called if the future resolves with an exception
+    f : Future object
+        Future to be monitored for exceptions
+    ignore : Exception or tuple of Exception
+        Exptected exception(s) to igore, i.e. they will not be logged.
```

Comment:
Fixed and pushed

On Thu, Feb 12, 2015 at 10:57 AM, profcalculus <notifications@github.com>
wrote:

> In katcp/core.py
> <https://github.com/ska-sa/katcp-python/pull/70#discussion_r24566260>:
>
> > @@ -97,6 +97,24 @@ def wrapped_coro(self, *args, **kwargs):
> >      return wrapped_coro
> >
> >  def log_future_exceptions(logger, f, ignore=()):
> > +    """Log any exceptions set to a future
> > +
> > +    Parameters
> > +    ----------
> > +    logger : logging.Logger instance
> > +        logger.exception(...) is called if the future resolves with an exception
> > +    f : Future object
> > +        Future to be monitored for exceptions
> > +    ignore : Exception or tuple of Exception
> > +        Exptected exception(s) to igore, i.e. they will not be logged.
>
> s/igore/ignore/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ska-sa/katcp-python/pull/70/files#r24566260>.
>